.github/workflows/gofmt.yml

```diff
@@ -0,0 +1,33 @@
+name: gofmt
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+    paths:
+      - "**/*.go"
+
+jobs:
+  run_gofmt:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
```

Comment:
Can we merge with the base branch before running this check?
https://github.com/GoogleCloudPlatform/magic-modules/blob/862a4608eae0216ab1a5300af9638d51e0b12ae4/.github/workflows/mmv1-check-templates.yml#L19-L27

Also can you pull the latest changes in from main?

Reply:
I added that in a new commit, and also rebased. Let me know if that looks right to you.

Reply:
This would basically have the effect of making sure it runs against new changes in `main` when the base is stale, but only in the case where a new commit is pushed, right?

_presumably_ if the check is already running on all other changes, I'd think there shouldn't be (m)any cases where this would be at issue, since the go files would already be have failed the check in their own branch if they were formatted poorly (and note that this check will only run if `.go` (vs. `.go.tmpl`) files are modified in a given PR. But it's not hard to add, and seems to work 👍

Reply:
Yeah correct.

It's more of an edge case where if we check in a change and this check is broken on a certain commit it's easy for use to fix forward for existing PRs without needing them to rebase.